### DISCUSSION TITLE
codeql cli results multiple languages

### DISCUSSION THREAD

**Discussion by sam-cha-tfs** [github/codeql#14597] [Q&A]
We have a repo that uses JS and C#. Our CI/CD uses the CLI to upload the results to GItHub but I don't see any notification if there were vulnerabilities on GitHub. The CLI outputs the sarif files were uploaded. I also only see the "last scanned" timestamp on GitHub. Below is a picture of the timestamp and commands I used. When I ran codeql locally I got vulnerabilities for one of the sarif files. Am I uploading correctly? What notifications do we receive from GitHub if there are no vulnerabilities? 

`codeql database create codeql-db --db-cluster --language=javascript,csharp --command=<build.pl>`

`codeql database analyze codeql-db/cpp cpp-security-extended.qls --format=sarif-latest --output=cpp.sarif --sarif-add-baseline-file-info`

`codeql database analyze codeql-db/javascript javascript-security-extended.qls --format=sarif-latest --output=js.sarif --sarif-add-baseline-file-info`

`codeql github upload-results --repository=<org/repo_name> --refs=ref/heads/master --sarif=cpp.sarif --commit=<commit id>`

`codeql github upload-results --repository=<org/repo_name> --refs=ref/heads/master --sarif=js.sarif --commit=<commit id>`

![image](https://github.com/github/codeql/assets/125404500/2ce357c8-5654-4b8b-9896-9319d7ed23ae)

**Comment by smowton** [comment]
I note that in https://docs.github.com/en/code-security/codeql-cli/getting-started-with-the-codeql-cli/analyzing-your-code-with-codeql-queries#running-codeql-database-analyze the blue box recommends using `--sarif-category` to distinguish the two different languages. Does including that option help?

**Reply by sam-cha-tfs** [reply]
I tried with the option and got the same results. Can this be because of the build script?

**Reply by smowton** [reply]
It's possible. For C# your build script needs to conduct a clean build of your code so we can follow along with the build process. For JavaScript the build command is meaningless. You may also want to check what happens if instead of building a multi-language database you just build two separate ones, specifying a build command just for the C# case.

**Reply by sam-cha-tfs** [reply]
Does the order of of the arguments matter? In the docs  `--commit` is before the `--sarif`

**Reply by smowton** [reply]
No. Assuming it still isn't working whether the databases are a multi-language cluster or individual databases, and including the --sarif-category argument, would you be happy to share sarif files for debugging purposes?

**Reply by sam-cha-tfs** [reply]
unfortunately I cannot share the sarif files. My guess is that it's similar to this issue below. The upload succeeds but there is no output from GitHub. My repo is fairly small so I don't think it is size related. Is there a way to debug the upload when using the cli tool? 

https://github.com/github/codeql/issues/8764

**Reply by smowton** [reply]
You can look at the SARIF using `jq` or anything else that parses JSON and verify that it contains results. Otherwise I would recommending analysing a toy project that will generate at least one hit, and which you would be able to share the project and/or its SARIF output so we can reproduce the problem.

**Reply by sam-cha-tfs** [reply]
Looks like I figured the issue. The file is uploaded to GitHub correctly. Alerts are only shown for the default branch as stated here in the link below. We would then need to filter the branch for the alerts. I can see alerts now but I still can't see the scan information. Is this because we don't have the scan enabled on the default branch? See my examples below where one repo has code scan enabled on the default branch and the other does not have it enabled on it's default branch.

https://docs.github.com/en/github-ae@latest/code-security/code-scanning/managing-code-scanning-alerts/managing-code-scanning-alerts-for-your-repository#viewing-the-alerts-for-a-repository

Fig 1.  The default branch does not have code scan enabled but the dev branch does. There is no "tool status"
![image](https://github.com/github/codeql/assets/125404500/3d74ce5e-50f6-4f8d-9a8d-17b051d3cb20)

Fig 2A. A repo where the default branch has code scan enabled which enables the feature branch "main-2" and shows "tool status"
![image](https://github.com/github/codeql/assets/125404500/be2a2042-81fa-437d-a0a7-c505d88d7711)

Fig 2B. I can click the tool status and view the scan results
![image](https://github.com/github/codeql/assets/125404500/cd35fd3e-562a-4b1e-953f-c770fe5d9caf)

**Reply by smowton** [reply]
Yes that's right, as per @jketema's response at https://github.com/orgs/community/discussions/61263

**Reply by smowton** [reply]
Since those are both JS results by the way, do check that use of sarif-category is enabling both C# and JS results to appear concurrently, perhaps by experimentally introducing a deliberate mistake in the C#.

**Reply by sam-cha-tfs** [reply]
Yes that's what I am unsure. I will check and experiment with it.

**Comment by felickz** [comment]
I know this is pseudo code but you are creating a charp DB and then attempting to analyze with cpp.  In practice you likely got an error where you already corrected that but FYI!
